### PR TITLE
Refactoring `save-orchestrator-common`

### PR DESCRIPTION
**Pull request by nulls**
`save-agent` needs to know only url to where need to post heartbeats and result of run

### CODE REVIEW

File: save-agent/src/linuxX64Main/kotlin/com/saveourtool/save/agent/SaveAgent.kt
Comment (nulls) [inline save-agent/src/linuxX64Main/kotlin/com/saveourtool/save/agent/SaveAgent.kt:161]: a single place where execution id is in used

File: save-agent/src/commonMain/kotlin/com/saveourtool/save/agent/utils/Utils.kt
Comment (petertrr) [inline save-agent/src/commonMain/kotlin/com/saveourtool/save/agent/utils/Utils.kt:11]: Same question here. For me it seems useful to know which versions of all components have been used for a particular execution to make debugging easier.

File: save-agent/src/commonMain/kotlin/com/saveourtool/save/agent/Main.kt
Comment (nulls) [reply save-agent/src/commonMain/kotlin/com/saveourtool/save/agent/Main.kt:57]: ok. will try to add it

File: save-agent/src/commonMain/kotlin/com/saveourtool/save/agent/utils/Utils.kt
Comment (nulls) [reply save-agent/src/commonMain/kotlin/com/saveourtool/save/agent/utils/Utils.kt:11]: here (in file name) version is not required

File: save-agent/src/commonMain/kotlin/com/saveourtool/save/agent/Main.kt
Comment (nulls) [reply save-agent/src/commonMain/kotlin/com/saveourtool/save/agent/Main.kt]: added as env variable

**Comment by akuleshov7** [comment]
You will have a lot of problems with conflicts resolving if you won't merge it, may be merge?

**Comment by nulls** [comment]
> You will have a lot of problems with conflicts resolving if you won't merge it, may be merge?

yes, I would like to merge. But I need an approval

File: save-backend/src/main/kotlin/com/saveourtool/save/backend/service/AgentService.kt
Comment (nulls) [inline save-backend/src/main/kotlin/com/saveourtool/save/backend/service/AgentService.kt:31]: need to refactor it to be ready to case when a single container can be used for different executions (long running agent)

**Review by akuleshov7 — APPROVED** [review]

**Pull request merged**
